core/lib/zksync_core/src/lib.rs

```diff
@@ -298,6 +298,50 @@ impl FromStr for Components {
     }
 }
 
+async fn ensure_l1_batch_commit_data_generation_mode(
+    state_keeper_config: &StateKeeperConfig,
+    contracts_config: &ContractsConfig,
+    eth_client: &impl EthInterface,
+) -> anyhow::Result<()> {
+    let selected_l1_batch_commit_data_generator_mode = state_keeper_config
+        .l1_batch_commit_data_generator_mode
+        .clone();
+    match get_pubdata_pricing_mode(&contracts_config, eth_client).await {
+        // Getters contract support getPubdataPricingMode method
+        Ok(l1_contract_pubdata_pricing_mode) => {
+            let l1_contract_batch_commitment_mode =
+                L1BatchCommitDataGeneratorMode::from_tokens(l1_contract_pubdata_pricing_mode)?;
```

Comment:
Small nit: it's usually a good idea to add [`context`](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#tymethod.context) to the errors you propagate. E.g. in this case you can do `.context("Unable to parse L1BatchCommitDataGeneratorMode received from L1 contract")?`.

This helps greatly when something happens, because by default you just see some bottommost error and you may even have no idea where it comes from (if there was no `context` added up the callstack).